### ISSUE TITLE
Xamarin Forms Casting Exception in LoadApplication

### ISSUE THREAD

**Issue by DumiduM** [seesharper/LightInject#464]
`var container = new ServiceContainer();

            container.Register<WardListPage>();
            container.Register<WardListViewModel>();
            container.Register<WardListService>();

            container.Register<PatientListViewPage>();
            container.Register<PatientListViewModel>();
            container.Register<PatientService>();

            container.Register<PatientDetailsViewPage>();
            container.Register<PatientDetailsViewModel>();

            var instance = container.GetInstance<WardListPage>();

            MainPage = new NavigationPage(instance);`

Getting exeption

**Comment by seesharper** [comment]
Could you provide a simple repro?

**Comment by jordansjones** [comment]
I don't understand what is happening here, but when registering a Xamarin.Forms page with the generic, `container.Register<SomePage>()`, vs using the factory, `container.Register(x => new SomePage(x.GetInstance<SomeViewModel>())` the resulting `ToolbarItems` property on the Page is converted from an `ObservableCollection<>` to a `List<>` which then causes the InvalidCastException. I will try to get a simple repro made.

**Comment by jordansjones** [comment]
Here is a reproducible example: https://github.com/jordansjones/LightInjectErrorExample
The main area of concern is here: https://github.com/jordansjones/LightInjectErrorExample/blob/de7334f17c67532acf37abc0a833c2632fbf1617/LightInjectErrorExample/LightInjectErrorExample/App.xaml.cs#L17-L34

**Comment by seesharper** [comment]
So it works when using the factory?

**Comment by jordansjones** [comment]
Yes. Meaning, the ObservableCollection isn't replaced with a List when using the factory.
